src/select-list-view.js

```diff
@@ -156,17 +156,38 @@ module.exports = class SelectListView {
 
   renderItems () {
     if (this.items.length > 0) {
+      if(!this.itemElementsPlain || this.itemElementsPlain && this.itemElementsPlain.size !== this.items.length) {
+        this.itemElementsPlain = new Map(this.items.map((item, index) => {
+          return [item, $(ListItemView, {
+            element: this.props.elementForItem(item, {selected: false, index}),
+            selected: false,
+            onclick: () => {
+              this.selectItem(item)
+              this.confirmSelection()
+            }
+          })]
+        }))
+        this.itemElementsSelected = new Map(this.items.map((item, index) => {
```

Comment:
I think these should probably be `WeakMap`s rather than maps so they don't cause us to hold on to items. That would just make it easier to not need to manually manage removing items from these collections to prevent leaking.

Rather than constructing the values up front, it seems like it would be better to lazily populate the maps whenever you grab an element. By constructing the elements up front for both maps, we're actually doing twice the work we need in the common case, since only one element is initially selected.

Reply:
`WeakMap` cannot have primitives as keys, so it made a number of tests break where `items` are arrays of primitives. However, the documentation says that `items` should be of type `[Object]`, so changing these test cases could be justified. Do you think it should be done in this PR?

Reply:
I do think we should go ahead and change the tests so we can use weak maps.

What are your thoughts on lazy populating?

Reply:
Lazy populating makes a lot of sense, and would be very easy to implement for the `selected` item element map, so I'm all for it in that case. For the plain elements I think it is best to just populate the map at once, since otherwise we would have to insert some kind of placeholder for the missing elements together with logic for figuring out when to actually render them. That becomes tricky because the consumers have a lot of freedom when creating the item elements.